ProcessSlave.php

```diff
@@ -384,18 +391,20 @@ protected function handleRequest(\React\Http\Request $request, HttpResponse $res
             );
             $this->shutdown();
         }
+        return $response;
     }
 
-    protected function prepareEnvironment(\React\Http\Request $request)
+    protected function prepareEnvironment(RequestInterface $request)
     {
         $_SERVER = $this->baseServer;
```

Comment:
Just read https://github.com/php-pm/php-pm/issues/199 again. Seems @marcj may have some other thoughts about how much abstraction for client applications is actually necessary. If more is the case we might still do this in some king of "legacy" adapter.